homeassistant/components/tile/__init__.py

```diff
@@ -1,137 +1,93 @@
 """The Tile component."""
 import asyncio
 from datetime import timedelta
+from functools import partial
 
 from pytile import async_login
 from pytile.errors import SessionExpiredError, TileError
 
-from homeassistant.const import ATTR_ATTRIBUTION, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import callback
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
 from homeassistant.helpers import aiohttp_client
-from homeassistant.helpers.update_coordinator import (
-    CoordinatorEntity,
-    DataUpdateCoordinator,
-    UpdateFailed,
-)
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
-from .const import DATA_COORDINATOR, DOMAIN, LOGGER
+from .const import DATA_COORDINATOR, DATA_TILE, DOMAIN, LOGGER
 
 PLATFORMS = ["device_tracker"]
 DEVICE_TYPES = ["PHONE", "TILE"]
 
-DEFAULT_ATTRIBUTION = "Data provided by Tile"
-DEFAULT_ICON = "mdi:view-grid"
 DEFAULT_UPDATE_INTERVAL = timedelta(minutes=2)
 
 CONF_SHOW_INACTIVE = "show_inactive"
 
 
 async def async_setup(hass, config):
     """Set up the Tile component."""
-    hass.data[DOMAIN] = {DATA_COORDINATOR: {}}
-
+    hass.data[DOMAIN] = {DATA_COORDINATOR: {}, DATA_TILE: {}}
     return True
 
 
-async def async_setup_entry(hass, config_entry):
+async def async_setup_entry(hass, entry):
     """Set up Tile as config entry."""
+    hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id] = {}
+    hass.data[DOMAIN][DATA_TILE][entry.entry_id] = {}
+
     websession = aiohttp_client.async_get_clientsession(hass)
 
-    client = await async_login(
-        config_entry.data[CONF_USERNAME],
-        config_entry.data[CONF_PASSWORD],
-        session=websession,
-    )
+    try:
+        client = await async_login(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            session=websession,
+        )
+        hass.data[DOMAIN][DATA_TILE][entry.entry_id] = await client.async_get_tiles()
+    except TileError as err:
```

Comment:
What if it fails because the internet connection was bad, should it retry by raising `ConfigEntryNotReady` ?